### ISSUE TITLE
InstallModuleError when training local on SageMaker notebook instance

### ISSUE THREAD

**Issue by mattmcclean** [aws/sagemaker-containers#137]
The latest version of `sagemaker-containers` is giving me an error when training my model locally on a SageMaker notebook instance. The error occurs when running the command: `/usr/local/bin/python -m pip install -U .`

The code block that failed in the notebook is the following:
```
from sagemaker.estimator import Estimator

hyperparameters = {'epochs': 4, 'batch-size': 64}

estimator = Estimator(role=role,
                      train_instance_count=1,
                      train_instance_type=instance_type,
                      image_name=image_name,
                      hyperparameters=hyperparameters)

estimator.fit(f'file://{DATA_PATH}')
```

Was working fine in a previous version of `sagemaker-containers`

Full stack trace can be found here: 

```
[{'DataSource': {'FileDataSource': {'FileDataDistributionType': 'FullyReplicated', 'FileUri': 'file:///home/ec2-user/SageMaker/sagemaker-fastai-example/data/shirts'}}, 'ChannelName': 'training', 'DataUri': 'file:///home/ec2-user/SageMaker/sagemaker-fastai-example/data/shirts'}]
Creating tmp4usrw_13_algo-1-VZHOB_1_12f99eb2c642 ... 
Attaching to tmp4usrw_13_algo-1-VZHOB_1_5c757e8ba99a2mdone
algo-1-VZHOB_1_5c757e8ba99a | 2018-11-19 23:17:56,515 sagemaker-containers INFO     Imported framework sagemaker_pytorch_container.training
algo-1-VZHOB_1_5c757e8ba99a | 2018-11-19 23:17:56,539 sagemaker_pytorch_container.training INFO     Block until all host DNS lookups succeed.
algo-1-VZHOB_1_5c757e8ba99a | 2018-11-19 23:17:56,542 sagemaker_pytorch_container.training INFO     Invoking user training script.
algo-1-VZHOB_1_5c757e8ba99a | 2018-11-19 23:17:56,542 sagemaker-containers INFO     Installing module with the following command:
algo-1-VZHOB_1_5c757e8ba99a | /usr/local/bin/python -m pip install -U . 
algo-1-VZHOB_1_5c757e8ba99a | Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
algo-1-VZHOB_1_5c757e8ba99a | 2018-11-19 23:17:56,969 sagemaker-containers ERROR    InstallModuleError:
algo-1-VZHOB_1_5c757e8ba99a | Command "/usr/local/bin/python -m pip install -U ."
tmp4usrw_13_algo-1-VZHOB_1_5c757e8ba99a exited with code 1
Aborting on container exit...
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/local/image.py in train(self, input_data_config, output_data_config, hyperparameters, job_name)
    124         try:
--> 125             _stream_output(process)
    126         except RuntimeError as e:

~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/local/image.py in _stream_output(process)
    556     if exit_code != 0:
--> 557         raise RuntimeError("Process exited with code: %s" % exit_code)
    558 

RuntimeError: Process exited with code: 1

During handling of the above exception, another exception occurred:

RuntimeError                              Traceback (most recent call last)
<ipython-input-12-08a4e64e55a4> in <module>()
      9                       hyperparameters=hyperparameters)
     10 
---> 11 estimator.fit(f'file://{DATA_PATH}')

~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/estimator.py in fit(self, inputs, wait, logs, job_name)
    207         self._prepare_for_training(job_name=job_name)
    208 
--> 209         self.latest_training_job = _TrainingJob.start_new(self, inputs)
    210         if wait:
    211             self.latest_training_job.wait(logs=logs)

~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/estimator.py in start_new(cls, estimator, inputs)
    460                                           resource_config=config['resource_config'], vpc_config=config['vpc_config'],
    461                                           hyperparameters=hyperparameters, stop_condition=config['stop_condition'],
--> 462                                           tags=estimator.tags)
    463 
    464         return cls(estimator.sagemaker_session, estimator._current_job_name)

~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/session.py in train(self, image, input_mode, input_config, role, job_name, output_config, resource_config, vpc_config, hyperparameters, stop_condition, tags)
    279         LOGGER.info('Creating training-job with name: {}'.format(job_name))
    280         LOGGER.debug('train request: {}'.format(json.dumps(train_request, indent=4)))
--> 281         self.sagemaker_client.create_training_job(**train_request)
    282 
    283     def tune(self, job_name, strategy, objective_type, objective_metric_name,

~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/local/local_session.py in create_training_job(self, TrainingJobName, AlgorithmSpecification, InputDataConfig, OutputDataConfig, ResourceConfig, **kwargs)
     72         training_job = _LocalTrainingJob(container)
     73         hyperparameters = kwargs['HyperParameters'] if 'HyperParameters' in kwargs else {}
---> 74         training_job.start(InputDataConfig, OutputDataConfig, hyperparameters, TrainingJobName)
     75 
     76         LocalSagemakerClient._training_jobs[TrainingJobName] = training_job

~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/local/entities.py in start(self, input_data_config, output_data_config, hyperparameters, job_name)
     68         self.state = self._TRAINING
     69 
---> 70         self.model_artifacts = self.container.train(input_data_config, output_data_config, hyperparameters, job_name)
     71         self.end = datetime.datetime.now()
     72         self.state = self._COMPLETED

~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/sagemaker/local/image.py in train(self, input_data_config, output_data_config, hyperparameters, job_name)
    128             # which contains the exit code and append the command line to it.
    129             msg = "Failed to run: %s, %s" % (compose_command, str(e))
--> 130             raise RuntimeError(msg)
    131 
    132         artifacts = self.retrieve_artifacts(compose_data, output_data_config, job_name)

RuntimeError: Failed to run: ['docker-compose', '-f', '/tmp/tmp4usrw_13/docker-compose.yaml', 'up', '--build', '--abort-on-container-exit'], Process exited with code: 1
```

**Comment by mattmcclean** [comment]
Forgot to add I am running `sagemaker-containers==2.3.1`

**Comment by laurenyu** [comment]
Thanks for the bug report, @mattmcclean! We have a fix out in #138 and will release it as soon as we can.

**Comment by mvsusp** [comment]
Hi @mattmcclean , we release 2.3.2 version with the fix https://pypi.org/project/sagemaker-containers/

Thanks for you feedback.
